git-link.el

```diff
@@ -471,6 +482,30 @@ return (FILENAME . REVISION) otherwise nil."
 
       (list host path))))
 
+(cl-defstruct ssh-host-info host resolved-host)
+
+(defun git-link--parse-ssh-config ()
+  "Parse ~/.ssh/config and return a list of Struct with host and hostname."
+  (let ((current-host nil)
+	(resolved-hostname nil)
+        (hosts nil))
+    (with-temp-buffer
+      (insert-file-contents "~/.ssh/config")
+      (dolist (line (split-string (buffer-string) "\n"))
+	(cond ((string-match "^Host[ ]\\(.*\\)" line)
```

Comment:
I wonder if this can be done with `tramp-get-connection-property`? I had a quick looksy and the first argument was a bit confusing so, maybe not?

Reply:
Yes, that function won't help. Infact, I initially looked into tramp to see if I can re-use any functions from there.

The closest I could find was `tramp-parse-sconfig`, but unfortunately that doesn't give information about the resolved hostname.

Reply:
Giving that an ssh config `Host can contain wildcards, thinking it makes sense to pass it through `ssh -G`. It is a bit of an edge case, but one upside is parsing its output seems easier. Thoughts?

Reply:
Yeah, I think that would be probably better. I will update this PR once I get some spare cycles.

Reply:
Updated implementation to use `ssh -G`.